ext/RMagick/rmpixel.c

```diff
@@ -1069,28 +1171,38 @@ Pixel_to_color(int argc, VALUE *argv, VALUE self)
         rb_raise(rb_eNoMemError, "not enough memory to continue.");
     }
 
+    exception = AcquireExceptionInfo();
+
     image->depth = depth;
-    image->matte = matte;
+#if defined(IMAGEMAGICK_6)
+    image->matte = alpha;
+#endif
 
     rm_init_magickpixel(image, &mpp);
     rm_set_magick_pixel_packet(pixel, &mpp);
 
-    exception = AcquireExceptionInfo();
-
     // Support for hex-format color names moved out of QueryMagickColorname
     // in 6.4.1-9. The 'hex' argument was removed as well.
     if (hex)
     {
         if (compliance == XPMCompliance)
         {
+#if defined(IMAGEMAGICK_7)
+            mpp.alpha_trait = UndefinedPixelTrait;
+#else
             mpp.matte = MagickFalse;
+#endif
             mpp.depth = (unsigned long) min(1.0 * image->depth, 16.0);
         }
         (void) GetColorTuple(&mpp, MagickTrue, name);
     }
     else
     {
+#if defined(IMAGEMAGICK_7)
+        (void) QueryColorname(image, &mpp, compliance, name, exception);
```

Comment:
Could we alias this, too?

Reply:
We can and I just pushed a patch for this.